### ISSUE TITLE
Setting handoff.port is dubious

### ISSUE THREAD

**Issue by macintux** [basho/riak_core#683]
```
$ ./bin/riak-admin set transfer_limit=50
Set transfer limit for 'dev1@127.0.0.1' to 50
$ ./bin/riak-admin set handoff.port=50
$
```

Seems like there should be some sanity checking around setting handoff port, or at the very least some confirmation/warning when you're done. (Also rather curious what it means to set the handoff port to a privileged port, but I don't see anything in the logs yet.)

**Comment by macintux** [comment]
And `set` command should indicate when a value is configured that won't survive a reboot.

**Comment by andrewjstone** [comment]
Regardless of whether we show it in handoff config, you can still attempt to set any config variable. None of them persist across reboot. We can't annotate or provide callbacks for all of them. That seems like a giant waste. What we could do is have a whitelist of settable config and disallow all other config from being set this way so we don't confuse users. I'm unclear if we can get this in in time for the build though.

**Comment by andrewjstone** [comment]
FYI All config variables do is set the app env in memory, unless a specific callback does something else. So if it's not a setting checked during runtime, setting it has no effect.

**Comment by andrewjstone** [comment]
Implemented the whitelist solution for now with #686
